bayesian_cnn_prometheus/analysis/correlation_analyzer.py

```diff
@@ -0,0 +1,159 @@
+            print(f"- peak signal to noise ratio: {self.metrics['peak_signal_noise_ratio']}")
+            # print(f"- mean structural similarity index: {self.metrics['ss_mean']}")
+
+    def _plot_metrics(self):
+        """
+        Plots all of the displayable metrics.
+        """
+        if self.metrics:
+            fig, axis = plt.subplots(1, 4)
+            axis[0].plot(self.metrics["contingency_table"])
+            axis[0].set_title("Contingency table")
+            axis[1].plot(self.metrics["ss_image"])
+            axis[1].set_title("Structural similarity image")
+            axis[2].plot(self.metrics["ss_gradient"])
+            axis[2].set_title("Structural similarity gradient")
+            axis[3].plot(self.metrics["cross_correlate"])
+            axis[3].set_title("Cross-correlate array")
+
+    def _assign_adapted_rand_error_metrics(self):
+        """
+        Computes Adapted Rand error, precision and recall.
+        """
+        self.metrics["are"], self.metrics["precision"], self.metrics["recall"] = \
+            metrics.adapted_rand_error(self.cancer_mask, self.variance_mask)
+
+    def _assign_variation_of_information_metrics(self):
+        """
+        Computes variation of information between cancer and variance masks.
+        """
+        self.metrics["cancer_to_variance_conditional_entropy"], self.metrics["variance_to_cancer_conditional_entropy"] = \
+            metrics.variation_of_information(self.cancer_mask, self.variance_mask)
+
+    def _assign_hausdorff_distance(self):
+        """
+        Computes the Hausdorff distance of nonzero cancer and variance masks voxels.
+        """
+        self.metrics["hausdorff_distance"] = metrics.hausdorff_distance(self.cancer_mask, self.variance_mask)
+
+    def _assign_mean_squared_error(self):
+        """
+        Computes Mean Squared Error between cancer and variance masks voxels.
+        """
+        self.metrics["mse"] = metrics.mean_squared_error(self.cancer_mask, self.variance_mask)
+
+    def _assign_normalized_root_mse(self):
+        """
+        Computes Normalized Root Mean Squared Error between cancer and variance masks voxels.
+        """
+        self.metrics["normalized_root_mse"] = metrics.normalized_root_mse(self.cancer_mask, self.variance_mask)
+
+    def _assign_peak_signal_noise_ratio(self):
+        """
+        Computes the peak signal to noise ratio between cancer and variance masks.
+        """
+        self.metrics["peak_signal_noise_ratio"] = metrics.peak_signal_noise_ratio(self.cancer_mask, self.variance_mask)
+
+    def _assign_structural_similarity_metrics(self):
+        """
+        Computes structural similarity metrics between cancer and variance masks.
+        """
+        self.metrics["ss_mean"], self.metrics["ss_gradient"], self.metrics["ss_image"] = \
+            metrics.structural_similarity(self.cancer_mask, self.variance_mask, gradient=True)
```

Comment:
Tutaj dostaję błąd, że są 2 zamiast 3 rzeczy w outpucie. Żeby był ten ss_image to ma być jeszcze parametr full=True

Reply:
Ale czy na pewno porzebujemy ten ss_image?

Reply:
Ogólnie te "displayable" metryki zostawiłbym jedynie w celach wizualizacji, a niekoniecznie porównywania modeli, także faktycznie może być on niepotrzebny